File: 0078-subsets/0078-subsets.py
class Solution(object):
    def subsets(self, nums):
        ret = []
        self.dfs(nums, [], ret)
        return ret
    
    def dfs(self, nums, path, ret):
        ret.append(path)
        for i in range(len(nums)):
            self.dfs(nums[i+1:], path+[nums[i]], ret)

File: 0078-subsets/0078-subsets.go
func subsets(nums []int) [][]int {
    res :=  [][]int{}
    solution(nums,[]int{},0,&res)
    return res
}

func solution(nums,current []int,index int, res *[][]int){
    
    *res = append(*res,current)
    for index < len(nums) {
        current = append(current,nums[index])
        solution(nums, current,index+1, res)
        current = append([]int{},current[:len(current)-1]...)
        index++
    }
}